core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingBlocksEndpoint.java

```diff
@@ -253,4 +259,34 @@ private static double getBlockOffset(int blockId, int trackChunkId, int trackSec
         throw new AssertionError(
                 String.format("getBlockOffset: Track chunk %s not in block %s", trackChunkId, blockId));
     }
+
+    private void validatePathfindingResult(PathfindingResult res, PathfindingWaypoint[][] reqWaypoints,
```

Comment:
We could also check that every track and route is covered from its start until its end (beside first and last ones).

I *suspect* it's not always the case and we'd miss it.